device/dhcp/clients/v4/dhcpv4client.yaml

```diff
@@ -0,0 +1,73 @@
+components:
+  schemas:
+    Device.Dhcpv4client:
+      description: >-
+        Configuration for emulated DHCPv4 Client on a single Interface.
+        https://www.rfc-editor.org/rfc/rfc2131.html
+      type: object
+      x-status:
+        status: under_review
+      required: [name]
+      properties:
+        name:
+          x-include: ../../../../common/common.yaml#/components/schemas/Named.Object/properties/name
+          x-field-uid: 1
+        selecting_server:
+          description: >-
+            The client receives one or more DHCPOFFER messages from one or more servers. The client may choose to wait for multiple responses.
+            The client chooses one server from which to request configuration
+            parameters, based on the configuration parameters offered in the DHCPOFFER messages.
+            use_first_server: if selected, the subnet accepts the IP addresses offered by the first server to respond with an offer of IP addresses.
+            use_server_address: The address of the DHCP server from which the subnet will accept IP addresses.
+            server address to 0.0.0.0, implementation will use the services of the first DHCP server that responds to the DHCP Lease Request message.
+          type: string
+          default: first_server
+          x-field-uid: 2
+          x-enum:
+            first_server:
+              x-field-uid: 1
+            server_address:
+              x-field-uid: 2
+        first_server:
+          type: string
+          format: ipv4
+          default: 0.0.0.0
+          x-field-uid: 3
+        server_address:
+          type: string
+          format: ipv4
+          default: 0.0.0.0
+          x-field-uid: 4
+        broadcast:
+          description: >-
+            If the broadcast bit is set, then the server and relay agent broadcast DHCPOFFER and DHCPACK messages.
+          type: boolean
+          default: false
+          x-field-uid: 5
+        gateway_ip: 
+          description: >-
+            A manually configured IPv4 gateway address. 
+            The default value 0.0.0.0, meaning that no manual gateway IP address is specified.
+            If the DHCPv4 Client receives from a server a response packet with a Router Address TLV 
+            specifying another gateway address than the manually configured gateway IP, 
+            then this TLV is ignored. 
+            Whenever a manual non default gateway IP is configured, the client will use it as gateway address 
+            for traffic outside of its network; 
+            otherwise client uses the gateway IP address learned from the server.
+          type: string
+          format: ipv4
+          default: 0.0.0.0
+          x-field-uid: 6
+        gateway_mac:
+          description: >-
+            A manually configured gateway MAC address. 
+            In case of non default, ARP messages are sent to resolve the gateway MAC address.
```

Comment:
Another way of doing this could be to make this an optional object . If included then use else treat as not configured ; instead of using default value to specify not configured.